src/lib/rengo_balancer.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2012-2022  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { errorAlerter } from "misc";
+import * as player_cache from "player_cache";
+import { put } from "requests";
+
+type Participant = {
+    user_id: number;
+    rating: number;
+};
+
+// Convert player IDs to Participants with ratings.
+async function toParticipants(playerIDs: number[]): Promise<Participant[]> {
+    const participants = [];
+    const required_fields = ["rating"];
+    for (const id of playerIDs) {
+        // TODO is there a way to fetch multiple players at once?
```

Comment:
I think `termination-api/players` supports this with the `ids` field, but `PlayerCache` may be doing this under the hood for you already with the `Batcher` utility:

https://github.com/online-go/online-go.com/blob/46b0852579d02da7035e6685f9b3cbf532395f93/src/lib/player_cache.ts#L241

You can confirm by watching the network tab in developer tools.

<img width="1416" alt="Screen Shot 2022-02-18 at 12 58 51 AM" src="https://user-images.githubusercontent.com/25233703/154651201-83fcd44d-bde0-4c5a-964f-bbb0504c9dc0.png">

Reply:
What is this witchcraft? Such nimble navigation 😮 starting to think you know the code better than anoek 😅